### DISCUSSION TITLE
Use Git Flow

### DISCUSSION THREAD

**Discussion by hishamco** [OrchardCMS/OrchardCore#14553] [Ideas]
As many know Git flow is a popular Git branching strategy, so can we use it to simplify the release management? There's a great integration with some of the Git Clients such as Fork, GitKraken .. etc

To be honest, I'm not sure how the release and hotfixes are done right now, but when I use it from Git Client it was so easy to manage release-related stuff

/cc @sebastienros @Piedone

**Comment by hishamco** [comment]
@MikeAlhayek your input will be valuable because you already did an experience with the release in some of the latest releases :)

**Comment by Piedone** [comment]
I think we're doing something very similar, or if you mean Git Flow in broad terms, that. Basically, there's the main development branch, `main`, with personal/issue-specific branches out of that, and release branches for a given release, as well as corresponding version tags. Except for having a separate, permanent branch to just merge to for releases (where we'd put the version tags) this sounds like Git Flow to me.

**Comment by sebastienros** [comment]
Yep, when we looked at it years ago it was what we were already doing. And it's working well IMO.

**Reply by hishamco** [reply]
I saw a great integration from within Fork git client to manage everything including releases. I hope to try it in `1.8.0` if you allow me to make the release

**Reply by Piedone** [reply]
What can it do? There's not much to manage for a release in Git.

**Reply by hishamco** [reply]
It just simplifies many things:

1. Creating features/hotfixes from `develop` branch
2.  Merge them later on `develop` 
3. Creating releases / tags
4. Merge on `main`

All of this from with the UI

**Reply by Piedone** [reply]
I'd be fine with testing doing our normal release process the next time with this, but I really don't see from what you describe what it can help, since there's so little to do (as in, git operations) already:

1. Open `release/x.y` branch from `main`.
2. Add version tag.
3. Merge to `main`.

Between #1 and #2 there might be hotfixes applied in separate branches, but what you do in git for that is also trivial.

**Reply by hishamco** [reply]
For those who already made a release, are you doing it from GitHub or Git Client / CLI

/cc @sebastienros @MikeAlhayek

**Reply by MikeAlhayek** [reply]
@hishamco creating a release in GitHub will trigger the workflow that pushes the release to nuget. So it's pretty streamlined process.

If you see there is an added value to change the current process, I suggest that you write a quick highlight of the current challenges. Additionally, suggest a solution by modifying the current process so solve the challenge. Otherwise, I see no reason to change the current process.

**Comment by hishamco** [comment]
I just created this discussion due to the last issues with `1.7.1` & `1.7.2`, Git Flow can easily manage and merge the hotfixes to the master and dev branches
